Add instruction count benchmark to run on pull requests

laithsakka opened the pull request:
This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```


Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131475



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

laithsakka committed:
test

[ghstack-poisoned]

laithsakka committed:
Update on "test"




[ghstack-poisoned]

laithsakka committed:
Update on "[DRAFT] test adding workflow "




[ghstack-poisoned]

laithsakka committed:
Update on "[DRAFT] test adding workflow "




[ghstack-poisoned]

laithsakka committed:
Update on "[DRAFT] test adding workflow "




[ghstack-poisoned]

laithsakka committed:
Update on "[DRAFT] test adding workflow "




[ghstack-poisoned]

laithsakka committed:
Update on "[DRAFT] test adding workflow "




[ghstack-poisoned]

laithsakka committed:
Update on "[DRAFT] test adding workflow "




[ghstack-poisoned]

laithsakka committed:
Update on "[DRAFT] test adding workflow "




[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull reuqests."

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
seems like we dont have access to hardware counters on the 
amazon machines that we use, only much larger machines provides that capability. 

I see one of our "Pull  time" tests using a 12X machine amz2023.linux.12xlarge, I will try to move the test to that one 
and check if it have access the arch_perfmon flag.

cat /proc/cpuinfo | grep arch_perfmon

https://oavdeev.github.io/posts/vpmu_support_z1d/
https://stackoverflow.com/questions/69395912/pmu-x86-64-performance-counters-not-showing-in-perf-under-aws

laithsakka committed:
Update on "[draft] Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
it worked on amaozn 12X machine 
![Uploading Screenshot 2024-07-27 at 12.12.30 PM.png…]()

laithsakka committed:
Update on "[draft] Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "[draft] Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "[draft] Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on master, following diffs will add checking out head~1 and 
running it and comparing. 




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

oulgen reviewed (commented):
Can you explain how the result is displayed on the PR? Is there an example PR that shows it?

How do we compare it against either base data or some hardcoded data?

laithsakka commented:
> Can you explain how the result is displayed on the PR? Is there an example PR that shows it?
> 
> How do we compare it against either base data or some hardcoded data?

I updated the summary to explain how to access the results, as mentioned in the summary this only run the benchmarks on this PR. next diffs will add the remaining features.
1. run on master
2. compare and fail on regression.
3. add comment to PR .

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results to to test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

kit1980 commented on .ci/pytorch/test.sh:
```diff
@@ -391,7 +392,21 @@ test_inductor_cpp_wrapper_abi_compatible() {
-if [[ "${TEST_CONFIG}" == *dynamo_eager* ]]; then
+pr_time_benchmarks() {
+
+  pip install cirron
```
Do you need to install cirron here as you already added it to requirements.txt ?

laithsakka replied:
When i did not have it here, it did failed with module not found

laithsakka commented:
I noticed that I am using XLA build, i will try to separate that

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
move test from XLA to linux-focal-cuda12_1-py3_10-gcc9-test

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

s7nfo commented:
👋 I'd suggest being explicit about the version of Cirron you want to install: `pip install cirron~=0.3` and `cirron ~= 0.3` in your `requirements.txt`. It's quite likely I'll introduce breaking changes in a new version at some point, but I'll keep 0.3.x compatible.

laithsakka commented on .github/workflows/pull.yml:
```diff
@@ -296,6 +295,7 @@ jobs:
       docker-image-name: pytorch-linux-focal-cuda12.1-cudnn9-py3-gcc9
       test-matrix: |
         { include: [
+          { config: "pr_time_benchmarks", shard: 1, num_shards: 1, runner: "${{ needs.get-label-type.outputs.label-type }}amz2023.linux.12xlarge.nvidia.gpu" },
```
interesting i did have some runs going through, 
I need a machine that have hardware counters and GPUs, 
I can use ones from the ones listed at  https://github.com/pytorch/test-infra/blob/main/.github/scale-config.yml 
do you have a suggestion on which one i shall use?
if the machine does not end with "nvidia.gpu" does it mean it does not have GPU ?

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
try linux.g4dn.12xlarge.nvidia.gpu

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
fix cirron version to 0.3

ezyang commented on .ci/pytorch/test.sh:
```diff
@@ -391,7 +392,21 @@ test_inductor_cpp_wrapper_abi_compatible() {
-if [[ "${TEST_CONFIG}" == *dynamo_eager* ]]; then
+pr_time_benchmarks() {
+
+  pip install cirron~=0.3
```
Let's not, please. The license is compatible so if you want to copy paste it in, that is fine by me, but a clean room implementation should also not be difficult.

laithsakka replied:
I am not quite sure I understand the comment, are complaining about the pip install or about fixing a specific version. can you clarify which one and why?
also what do you mean by  "clean room implementation"?

ezyang replied:
I think taking on an extra third party dependency just to get instruction counts is not a good trade. Although there are some mitigating factors, e.g., we would only be using this during testing, we plan to be using the instruction counts durably over time, and I don't want to deal with "oops, cirron updated and now the instruction counts are all different and our numbers are all wrong now". It's a small enough amount of code that we should take ownership of it, properly review it, and then patch it if there are problems.

(cc @malfet)

ezyang replied:
Clean room implementation here means implementing the dep from scratch, without referencing the original's source code, which is something you would do if it was incompatibly licensed.

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
avoid depending on external cirron.

kit1980 commented on benchmarks/dynamo/pr_time_benchmarks/benchmark_base.py:
```diff
@@ -0,0 +1,62 @@
+import csv
+from abc import ABC, abstractmethod
+
+from benchmarks.dynamo.pr_time_benchmarks.cirron.cirron import Collector
```
I don't see benchmarks.dynamo.pr_time_benchmarks.cirron.cirron ?

Also if you copy cirron, you should probably copy into third_party, see https://github.com/pytorch/pytorch/tree/main/third_party/miniz-2.1.0 for example.
And make sure all the licenses are in place.

laithsakka replied:
looks like i forgot to add it, question if i copy it there would it not sync? I am also modifying it alot just copying two files from it

laithsakka commented:
add missing files to commit

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

ezyang commented on benchmarks/dynamo/pr_time_benchmarks/cirron/cirron.py:
```diff
@@ -0,0 +1,108 @@
+    if exit_status != 0:
+        raise Exception(
+            "Failed to compile cirronlib.cpp, make sure you have 'c++' installed."
+        )
```
wth is this lol. Use cpp_extension or better yet put it in the main torch/csrc

laithsakka replied:
This is how cirron.py is implemented originally I ported as is, I talked to 
@aorenste and he gave me pointers on how torch handles c++ functions i guess I will create new functions
under torch._C and call them. probably we will doing a "clean room implementation" at that point

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
avoid using cirron and build a local instruction counter

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
using amz2023.linux.12xlarge will have a separate diff in case we need to change it, will have a seperate discussion about that seperalatley

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
having the test as part of linux-focal-py3_8-clang9-xla-build might not be the most ideal but that I will address on a different commit after all those land, the test runs fine on linux-focal-py3_8-clang9-xla-build/amz2023.linux.12xlarge. for now

laithsakka commented:
FYI I am expirmenting with the machines to run the tests on but the PR is ready for review.

ezyang commented on torch/csrc/instruction_counter/Module.cpp:
```diff
@@ -0,0 +1,87 @@
+  auto instruction_counter = m.def_submodule(
+      "_instruction_counter", "instruction_counter related pybind.");
+  instruction_counter.def("start", start);
+  instruction_counter.def("end", end);
```
You should seriously consider doing the binding manually by hand, pybind11 adds quite a lot of overhead. You should be able to measure exactly what that overhead is with your fancy new instruction counter

laithsakka replied:
I will visit that in a later diff, since the overhead should be consistent its not a very big issue for the goal of detecting regressions (well unless it was much higher than the benchmark lol) I will try to measure and address if needed in different PR

ezyang commented on torch/csrc/instruction_counter/Module.cpp:
```diff
@@ -0,0 +1,87 @@
+long start() {
+#if !defined(__linux__)
+  printf("This systems seems not to be Linux");
+  return -1;
```
Raise an exception instead of returning -1?

ezyang reviewed (approved):
The code part looks fine. What's the plan for doing something useful with the instruction count numbers?

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
> The code part looks fine. What's the plan for doing something useful with the instruction count numbers?

The two other PRs in the stack run this on Head~1 and compare results to detect regressions. 
1. next we should add more benchmarks. 
2. maybe we can upload results for historical view.

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
using linux.g4dn.metal.nvidia.gpu for now which have both GPU and hardware counters.

laithsakka commented:
@pytorchbot merge

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka committed:
Update on "Add instruction count benchmark to run on pull requests"

This PR only adds the execution of the benchmarks on this PR and print results, following diffs will add checking out head~1 and running it and comparing. 

to access results goto test pr_time_benchmarks and inspect logs:
you should see
```
+ echo 'benchmark results on current PR: '
benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,27971461254
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge
